docs/source/api/rasterio-udfs.rst

```diff
@@ -234,40 +234,118 @@ Firstly we will create a spark DataFrame from a directory of raster files.
     +-----------------------------------------------------------+------------------------------+-----------+---------------------+-------+-------+-----------+----------------------+-------------+-------+---------------------------------------------------------------------------------------------------------------+
 
 
-Next we will define a function that will write a given raster file to disk.
+Next we will define a function that will write a given raster file to disk. A "gotcha" to keep in mind is that you do
+not want to have a file context manager open when you go to write out its context as the context manager will not yet
+have been flushed.
 
 .. code-block:: python
 
-    import numpy as np
-    import rasterio
-    from rasterio.io import MemoryFile
-    from io import BytesIO
-    from pyspark.sql.functions import udf
-    from pathlib import Path
+    @udf("string")
+    def write_raster(raster, driver, file_id, fuse_dir):
+        from io import BytesIO
+        from pathlib import Path
+        from pyspark.sql.functions import udf
+        from rasterio.io import MemoryFile
+        import numpy as np
+        import rasterio
+        import shutil
+        import tempfile
+
+        # - [1] populate the initial profile
+        # # profile is needed in order to georeference the image
+        profile = None
+        with MemoryFile(BytesIO(raster)) as memfile:
+            with memfile.open() as dataset:
+              profile = dataset.profile
+
+        # - [2] get the correct extension
+        extensions_map = rasterio.drivers.raster_driver_extensions()
```

Comment:
@mjohns-databricks this feels like something we could add to mosaic as a utility function.
I think mosaic.util would be a good idea.